### PR TITLE
optimization in hypre_CSRMatrixBigJtoJ and JtoBigJ

### PR DESCRIPTION
**Pull request by liruipeng**
Code optimization as suggested in #132. @v-dobrev can you also take a look? Thanks!

### CODE REVIEW

**Review by rfalgout — APPROVED** [review]
Looks good to me!  -Rob

File: src/seq_mv/csr_matrix.c [hunk 1]
```diff
@@ -215,19 +216,23 @@ hypre_CSRMatrixBigJtoJ( hypre_CSRMatrix *matrix )
    HYPRE_Int     num_nonzeros = hypre_CSRMatrixNumNonzeros(matrix);
    HYPRE_BigInt *matrix_big_j = hypre_CSRMatrixBigJ(matrix);
    HYPRE_Int    *matrix_j = NULL;
-   HYPRE_Int     i;
+   HYPRE_Int     ierr = 0;
```
Comment (victorapm) [inline src/seq_mv/csr_matrix.c]: This is outside the purpose of this PR, but should we do `return ierr` when the value of `ierr` can't be changed in a function? How about replacing it to `return hypre_error_flag`? My understanding is that the latter approach is preferred so I wanted to check this with everyone :)

**Review by victorapm — APPROVED** [review]
Looks good to me, thanks!

**Review by oseikuffuor1 — APPROVED** [review]
Looks good to me, thanks. I also agree with Victor that we should use hypre_error_flag instead of ierr, if ierr is not used. It might even throw a compiler warning if the unused-variable warning option is turned on, which would fail the regression test.

**Review by ulrikeyang — APPROVED** [review]

**Comment by liruipeng** [comment]
Made changes per suggestions from @victorapm @oseikuffuor1. `machine-tux.sh` was clean.

**Pull request merged**